pkg/util/serviceurl.go

```diff
@@ -0,0 +1,12 @@
+package util
+
+import (
+	"crypto/sha256"
+	"fmt"
+)
+
+func GenerateServiceURL(cluster, namespace, service string) string {
+	// Using hash to generate a random string; URL string should be less than 63 characters
+	// Keep namespace and service for debugging purpose
+	return fmt.Sprintf("%s-%s-%x", namespace, service, sha256.Sum256([]byte(fmt.Sprintf("%s_%s_%s", cluster, namespace, service))))[:(63 - len(namespace) - len(service) - 1)]
```

Comment:
what does `:(63 - len(namespace) - len(service) - 1)` mean? how do you ensure it is positive?

Reply:
Yeah, that's a bug, it should be a fixed prefix.

Reply:
Updated.